On most LCDs (at least Sharp's one) LCD logic needs to be activated before
powering on the backlight, otherwise some display's startup artefacts 
will be seen.

Signed-off-by: Nicolas Colombain <nicolas.colombain@armadeus.com>
Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>

Index: linux-2.6.36/drivers/video/imxfb.c
===================================================================
--- linux-2.6.36.orig/drivers/video/imxfb.c	2010-10-20 22:30:22.000000000 +0200
+++ linux-2.6.36/drivers/video/imxfb.c	2010-11-25 17:46:43.000000000 +0100
@@ -542,10 +542,10 @@
 
 	clk_enable(fbi->clk);
 
-	if (fbi->backlight_power)
-		fbi->backlight_power(1);
 	if (fbi->lcd_power)
 		fbi->lcd_power(1);
+	if (fbi->backlight_power)
+		fbi->backlight_power(1);
 }
 
 static void imxfb_disable_controller(struct imxfb_info *fbi)
